### PR TITLE
Add latitude and longitude values to datastore

### PR DESCRIPTION
**Pull request by winniewan**

### CODE REVIEW

File: capstone/src/main/java/com/google/sps/servlets/authentication/NewUserServlet.java [hunk 1]
```diff
@@ -60,7 +65,9 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       Entity userEntity = new Entity(PROFILE_TASK_NAME, userId);
       userEntity.setProperty(IS_BUSINESS_PROPERTY, NO);
       userEntity.setProperty(NAME_PROPERTY, ANONYMOUS_NAME);
-      userEntity.setProperty(LOCATION_PROPERTY, NULL_STRING);
+      userEntity.setProperty(LOCATION_PROPERTY, DEFAULT_LOCATION);
+      userEntity.setProperty(LAT_PROPERTY, DEFAULT_LAT);
+      userEntity.setProperty(LONG_PROPERTY, DEFAULT_LONG);
```
Comment (lukasberglund) [inline capstone/src/main/java/com/google/sps/servlets/authentication/NewUserServlet.java:70]: Also, maybe we can create a location object and store it like that? I.e. you create a Location.java file that stores Lat, Long, and Text Location

File: capstone/src/main/webapp/js/profile.js [hunk 2]
```diff
@@ -86,15 +86,46 @@ function redirectToDefaultPage(id) {
...
+  var geocoder = new google.maps.Geocoder();
+
+  return new Promise((resolve, reject) => {
+    geocoder.geocode(request, function(results, status) {
+      if (status == google.maps.GeocoderStatus.OK) {
+        resolve(results);
+      } else {
+        reject();
+      }
+    })
```
Comment (lukasberglund) [inline capstone/src/main/webapp/js/profile.js:126]: What's the use in returning this as a promise?

File: capstone/src/main/webapp/js/profile.js [hunk 3]
```diff
@@ -86,15 +86,46 @@ function redirectToDefaultPage(id) {
...
+  })
+}
+
+// Convert address to latitude and longitude values.
+async function convertToLatLong() {
+  let address = document.getElementById('edit-location').value;
+  let lat = document.getElementById('edit-lat');
+  let long = document.getElementById('edit-long');
+
+  var geocoder = new google.maps.Geocoder();
```
Comment (lukasberglund) [inline capstone/src/main/webapp/js/profile.js:109]: Why not use `const` or `let` here?

File: capstone/src/test/java/com/google/sps/servlets/profile/ProfileServletTest.java [hunk 4]
```diff
@@ -271,13 +268,33 @@ public void userEditProfileAddToDatastore() throws Exception {
...
 
     profileServlet.doPost(request, response);
 
     // verify if a sendError() was performed with the expected values.
     Mockito.verify(response, Mockito.times(1))
         .sendError(Mockito.eq(HttpServletResponse.SC_NOT_FOUND), Mockito.anyString());
   }
+
+  // Helper function to set business entity properties.
+  public Entity setBusinessEntityProperty() {
```
Comment (lukasberglund) [inline capstone/src/test/java/com/google/sps/servlets/profile/ProfileServletTest.java:281]:
```suggestion
  private Entity setBusinessEntityProperty() {
```

File: capstone/src/test/java/com/google/sps/servlets/profile/ProfileServletTest.java [hunk 5]
```diff
@@ -271,13 +268,33 @@ public void userEditProfileAddToDatastore() throws Exception {
...
+    ent.setProperty(LOCATION_PROPERTY, LOCATION);
+    ent.setProperty(LAT_PROPERTY, LAT);
+    ent.setProperty(LONG_PROPERTY, LONG);
+    ent.setProperty(BIO_PROPERTY, BIO);
+
+    return ent;
+  }
+
+  // Helper function to set getParameter() method.
+  public void setRequestParameters() {
```
Comment (lukasberglund) [inline capstone/src/test/java/com/google/sps/servlets/profile/ProfileServletTest.java:294]:
```suggestion
  private void setDefaultRequestParameters() {
```

File: capstone/src/main/webapp/profile.html [hunk 6]
```diff
@@ -90,7 +95,7 @@ <h3 id="bio"></h3>
             </div>
           </div>
 
-          <input type="submit" value="Submit" id="submit-button" onclick="submitProfileForm()">
+          <input type="button" value="Submit" id="submit-button" onclick="submitProfileForm()">
```
Comment (eash130) [inline capstone/src/main/webapp/profile.html:98]: Why switch to button type?

File: capstone/src/main/java/com/google/sps/servlets/authentication/NewUserServlet.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (winniewan) [reply capstone/src/main/java/com/google/sps/servlets/authentication/NewUserServlet.java:70]: Yep, thanks for the input! I will change it to null or none by default

Comment (winniewan) [reply capstone/src/main/java/com/google/sps/servlets/authentication/NewUserServlet.java:70]: Great suggestion! I will create a GeoPt for combining Lat Long instead

File: capstone/src/test/java/com/google/sps/servlets/profile/ProfileServletTest.java (hunk 4 above)
Comment (winniewan) [reply capstone/src/test/java/com/google/sps/servlets/profile/ProfileServletTest.java:281]: done

File: capstone/src/test/java/com/google/sps/servlets/profile/ProfileServletTest.java (hunk 5 above)
Comment (winniewan) [reply capstone/src/test/java/com/google/sps/servlets/profile/ProfileServletTest.java:294]: done

File: capstone/src/main/webapp/js/profile.js (hunk 3 above)
Comment (winniewan) [reply capstone/src/main/webapp/js/profile.js:109]: I'll use let. done.

File: capstone/src/main/webapp/js/profile.js (hunk 2 above)
Comment (winniewan) [reply capstone/src/main/webapp/js/profile.js:127]: By returning the Promise where the resolve() of that promise is called inside the callback passed to geocoder.geocode(), it makes sure that the callback is called before the form submits so it's entered in the datastore.

File: capstone/src/main/webapp/profile.html (hunk 6 above)
Comment (winniewan) [reply capstone/src/main/webapp/profile.html:98]: So I can geocode the latitude and longitude before the form submits by using promises and async/await

**Pull request merged**